packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/DbSyncNetworkInfoProvider.ts

```diff
@@ -112,6 +112,7 @@ export class DbSyncNetworkInfoProvider extends DbSyncProvider implements Network
 
   async close(): Promise<void> {
     this.#cache.shutdown();
+    await this.#genesisDataReady;
```

Comment:
Just of a question: why execute the genesis data promise in close()?

Reply:
The `Promise` is created in the `constructor` and `await`ed in `genesisParameters` and `lovelaceSupply` methods.

In case `loadGenesisData` fails and `shutdown` is called without never calling neither `genesisParameters` nor `lovelaceSupply`, an unhandled rejection error is raised.